File: Python/1021.py
valor = float(input())
qntd = 0
print("NOTAS:")
while valor >= 100:
    qntd += 1
    valor -= 100
print(qntd, "nota(s) de R$ 100.00")
qntd = 0
while valor >= 50:
    qntd += 1
    valor -= 50
print(qntd, "nota(s) de R$ 50.00")
qntd = 0
while valor >= 20:
    qntd += 1
    valor -= 20
print(qntd, "nota(s) de R$ 20.00")
qntd = 0
while valor >= 10:
    qntd += 1
    valor -= 10
print(qntd, "nota(s) de R$ 10.00")
qntd = 0
while valor >= 5:
    qntd += 1
    valor -= 5
print(qntd, "nota(s) de R$ 5.00")
qntd = 0
while valor >= 2:
    qntd += 1
    valor -= 2
print(qntd, "nota(s) de R$ 2.00")
qntd = 0
print("MOEDAS:")
while valor >= 1:
    qntd += 1
    valor -= 1
print(qntd, "moeda(s) de R$ 1.00")
qntd = 0
while valor >= 0.5:
    qntd += 1
    valor -= 0.5
print(qntd, "moeda(s) de R$ 0.50")
qntd = 0
while valor >= 0.25:
    qntd += 1
    valor -= 0.25
print(qntd, "moeda(s) de R$ 0.25")
qntd = 0
while valor >= 0.1:
    qntd += 1
    valor -= 0.1
print(qntd, "moeda(s) de R$ 0.10")
qntd = 0
while valor >= 0.05:
    qntd += 1
    valor -= 0.05
print(qntd, "moeda(s) de R$ 0.05")
qntd = round(valor * 100)
print(qntd, "moeda(s) de R$ 0.01")

File: Javascript/1021.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

let dinheiro = parseFloat(lines.shift());
console.log("NOTAS:")
console.log(Math.floor(dinheiro / 100), "nota(s) de R$ 100.00");
dinheiro = dinheiro % 100;
console.log(Math.floor(dinheiro / 50), "nota(s) de R$ 50.00");
dinheiro = dinheiro % 50;
console.log(Math.floor(dinheiro / 20), "nota(s) de R$ 20.00");
dinheiro = dinheiro % 20;
console.log(Math.floor(dinheiro / 10), "nota(s) de R$ 10.00");
dinheiro = dinheiro % 10;
console.log(Math.floor(dinheiro / 5), "nota(s) de R$ 5.00");
dinheiro = dinheiro % 5;
console.log(Math.floor(dinheiro / 2), "nota(s) de R$ 2.00");
dinheiro = dinheiro % 2;
console.log("MOEDAS:")
console.log(Math.floor(dinheiro), "moeda(s) de R$ 1.00");
dinheiro = dinheiro % 1;
console.log(Math.floor(dinheiro / 0.5), "moeda(s) de R$ 0.50");
dinheiro = dinheiro % 0.5;
console.log(Math.floor(dinheiro / 0.25), "moeda(s) de R$ 0.25");
dinheiro = dinheiro % 0.25;
console.log(Math.floor(dinheiro / 0.1), "moeda(s) de R$ 0.10");
dinheiro = dinheiro % 0.1;
console.log(Math.floor(dinheiro / 0.05), "moeda(s) de R$ 0.05");
dinheiro = dinheiro % 0.05;
console.log(Math.round(dinheiro / 0.01), "moeda(s) de R$ 0.01");
